weave

```diff
@@ -804,6 +816,51 @@ dns_args() {
     fi
 }
 
+default_etc_hosts() {
+    name=$1
+    shift
+    extra_hosts=$@
+    echo "# created by Weave - BEGIN"
+    echo "# container hostname"
+    for cidr in $ALL_CIDRS ; do
+        echo "${cidr%/*}    $name"
+    done
+    echo
+    echo "# static names added with --add-host"
+    for host in $extra_hosts ; do
+        name=$(echo $host | cut -f1 -d":")
+        ip=$(echo $host | cut -f2 -d":")
+        echo "$ip     $name"
+    done
+
+    cat <<-EOF
+
+# default localhost entries
+127.0.0.1       localhost
+::1             ip6-localhost ip6-loopback
+fe00::0         ip6-localnet
+ff00::0         ip6-mcastprefix
+ff02::1         ip6-allnodes
+ff02::2         ip6-allrouters
+EOF
+
+    echo "# created by Weave - END"
+}
+
+rewrite_etc_hosts() {
+    hosts_path_and_name=$(docker inspect -f '{{.HostsPath}} {{.Config.Hostname}}' $CONTAINER)
+    hosts=$(echo $hosts_path_and_name | cut -f1 -d" ")
+    name=$(echo $hosts_path_and_name | cut -f2 -d" ")
+    containers_path=$(dirname $hosts)
+    mnt=/container
+    mnt_hosts=$mnt/$(basename $hosts)
+    contents="$(default_etc_hosts $name $@)"
+    docker run --rm \
+        -v $containers_path:$mnt \
+        --entrypoint=sh \
+        $EXEC_IMAGE -c "echo '$contents' > $mnt_hosts && rm -f $mnt_hosts && echo '$contents' > $mnt_hosts"
```

Comment:
It mimics the behaviour described by @awh [here](https://github.com/weaveworks/weave/issues/1374#issuecomment-140118011). I think it could be removed...